tests/cli/main.cpp

```diff
@@ -1409,3 +1410,184 @@ BOOST_FIXTURE_TEST_CASE( general_storage, cli_fixture )
+      });
+
+
+      //////
+      // Alice authorizes Bob to transfer funds from her account to Charlie's account
+      //////
+      graphene::wallet::transaction_handle_type tx_alice_handle = con_alice.wallet_api_ptr->begin_builder_transaction();
+
+      custom_authority_create_operation caop;
+      caop.account = alice_acc.get_id();
+      caop.auth.add_authority(bob_acc.get_id(), 1);
+      caop.auth.weight_threshold = 1;
+      caop.enabled = true;
+      caop.valid_to = db->head_block_time() + 1000;
+      caop.operation_type = operation::tag<transfer_operation>::value;
+
+      // Restriction should have "to"  equal Charlie
+      vector<restriction> restrictions;
+      auto to_index = member_index<transfer_operation>("to");
+      restrictions.emplace_back(to_index, restriction::func_eq, charlie_acc.get_id());
+
+      con_alice.wallet_api_ptr->add_operation_to_builder_transaction(tx_alice_handle, caop);
+      asset ca_fee = con_alice.wallet_api_ptr->set_fees_on_builder_transaction(tx_alice_handle, GRAPHENE_SYMBOL);
+
+      // Sign the transaction with the implied nathan's key and the explicitly yet unnecessary Bob's key
+      signed_trx = con_alice.wallet_api_ptr->sign_builder_transaction2(tx_alice_handle, {}, true);
+
+      // Check for one signatures on the transaction
+      BOOST_CHECK_EQUAL(signed_trx.signatures.size(), 1);
+
+      // Check that the signed transaction contains Bob's signature
+      BOOST_CHECK_EQUAL(nathan_acct.active.get_keys().size(), 1);
+      flat_set<public_key_type> expected_signers = {alice_bki.pub_key};
+      flat_set<public_key_type> actual_signers = con_alice.wallet_api_ptr->get_transaction_signers(signed_trx);
+      BOOST_CHECK(actual_signers == expected_signers);
+
+
+      //////
+      // Initialize Bob's CLI wallet
+      //////
+      client_connection con_bob(app1, app_dir, server_port_number, "wallet_bob.json");
+      con_bob.wallet_api_ptr->set_password("supersecret");
+      con_bob.wallet_api_ptr->unlock("supersecret");
+
+      // Import Bob's key
+      BOOST_CHECK(con_bob.wallet_api_ptr->import_key("bob", bob_bki.wif_priv_key));
+
+
+      //////
+      // Bob attempt to transfer funds from Alice to Charlie while using Bob's wallet
+      // This should succeed because Bob is authorized to transfer by Alice
+      //////
+      graphene::wallet::transaction_handle_type tx_bob_handle = con_bob.wallet_api_ptr->begin_builder_transaction();
+
+      const asset transfer_amount = asset(123 * GRAPHENE_BLOCKCHAIN_PRECISION);
+      transfer_operation top;
+      top.from = alice_acc.id;
+      top.to = charlie_acc.id;
+      top.amount = transfer_amount;
+
+      con_bob.wallet_api_ptr->add_operation_to_builder_transaction(tx_bob_handle, top);
+      asset transfer_fee = con_bob.wallet_api_ptr->set_fees_on_builder_transaction(tx_bob_handle, GRAPHENE_SYMBOL);
+
+      // Sign the transaction with the implied nathan's key and the explicitly yet unnecessary Bob's key
```

Comment:
I think you mean Alice's key is implied (but not available). Bob's key is required.